tty/uart16550/uart16550.c

```diff
@@ -76,15 +76,15 @@ static void set_baudrate(void *_uart, speed_t baud)
 {
 	uint8_t reg;
 	uart_t *uart = (uart_t *)_uart;
-	int baud_rate = libtty_baudrate_to_int(baud);
-
-	if (baud_rate <= 0 || baud_rate > 115200) {
-		return;
-	}
+	int32_t baud_rate = libtty_baudrate_to_int(baud);
 
 	/* Baud divisor */
 	baud_rate = uart->clk / (16 * baud_rate);
 
+	if (baud_rate <= 0 || baud_rate > UINT16_MAX) {
```

Comment:
Baudrate < 0 should be checked before regval calculation. 0 is a valid value to write (baudrate == 0 is not ofc)

Reply:
Yes, you`re right. To add more - `libtty_baudrate_to_int(B0)` returns 0 so it would crash the driver (I didn't expect B0 to be even possible)